src/wwwroot/services/settings.js

```diff
@@ -56,14 +64,22 @@
       });
     }
 
-    function deleteDomain (domainName) {
+    function deleteDomain (domainName, onExpectedError) {
+      var tryHandleError = function (rejection) {
```

Comment:
It seems to be duplicated, do you think that it is possible to remove code duplication?